py/bitbox02/bitbox02/communication/bitbox_api_protocol.py

```diff
@@ -232,13 +256,20 @@ def _create_noise_channel(self, noise_config: BitBoxNoiseConfig) -> NoiseConnect
         noise.set_keypair_from_private_bytes(Keypair.STATIC, private_key)
         noise.set_prologue(b"Noise_XX_25519_ChaChaPoly_SHA256")
         noise.start_handshake()
-        noise.read_message(self._query(noise.write_message()))
+        start_handshake_status, start_handshake_reply = self._handshake_query(noise.write_message())
+        if start_handshake_status != RESPONSE_SUCCESS:
```

Comment:
similar to the other comment, this code could also be moved to `_handshake_query` of `BitBoxProtocolV7`. Then `handshake_query` does not have to return the status code, and avoid the "fake" response in V1-V6 (`return RESPONSE_SUCCESS`).

Reply:
This allows to throw with the correct error message (as `_handshake_query` doesn't know "which" part of the handshake has failed)